leakcanary-analyzer/src/main/java/com/squareup/leakcanary/HahaHelper.java

```diff
@@ -66,7 +66,7 @@ static boolean extendsThread(ClassObj clazz) {
     boolean extendsThread = false;
     ClassObj parentClass = clazz;
     while (parentClass.getSuperClassObj() != null) {
```

Comment:
This check also seems wrong. It should probably be `parentClass != null`. It still works because in the case where a class extends `Thread`, when we reach `Thread` as `parentClass` it still has a super class of `Object`.

Reply:
Yes, you are right, thank you.